src/hal/components/lcd.c

```diff
@@ -279,6 +278,8 @@ static void write_one(lcd_inst_t *inst){
         return;
     }    
     
+    if (*inst->page_num >= inst->num_pages) return; // should this error?
+
```

Comment:
you'd move this the
~~~~
 if (*inst->contrast != inst->last_contrast){
~~~~
above this early-return if you wanted to track contrast changes while displaying an invalid page.  This is safe, as that if-block doesn't refer to inst->pages[*inst->page_num].